src/Columns/ActionColumn.php

```diff
@@ -114,94 +89,103 @@ class ActionColumn extends Column
      *               The signature of the callback should be the same as that of [[createUrl()]]
      *               Since 2.0.10 it can accept additional parameter, which refers to the column instance itself:
-     *
      * ```php
      * function (string $action, mixed $model, mixed $key, integer $index, ActionColumn $this) {
      *     //return string;
      * }
      * ```
-     *
      * If this property is not set, button URLs will be created using [[createUrl()]].
      */
     public $urlCreator;
     /**
      * @var array html options to be applied to the [[initDefaultButton()|default button]].
-     *
      * @since 2.0.4
      */
     public $buttonOptions = [];
 
-    /**
-     * {@inheritdoc}
-     */
-    public function __construct(string $template = null)
-    {
-        if ($template !== null) {
-            $this->template = $template;
-        }
-        $this->initDefaultButtons();
-    }
-
     /**
      * Initializes the default button rendering callbacks.
      */
-    protected function initDefaultButtons()
+    public function init(): self
     {
         $this->initDefaultButton('view', 'eye-open');
         $this->initDefaultButton('update', 'pencil');
-        $this->initDefaultButton('delete', 'trash', [
-            'data-confirm' => Yii::t('yii', 'Are you sure you want to delete this item?'),
-            'data-method'  => 'post',
-        ]);
+        // TODO fix
+        $language = 'language';
+        $this->initDefaultButton(
+            'delete',
+            'trash',
+            [
+                'data-confirm' => self::$messageFormatter->format(
+                    'Are you sure you want to delete this item?',
+                    [],
+                    $language
+                ),
+                'data-method' => 'post',
+            ]
+        );
+
+        return $this;
     }
 
     /**
      * Initializes the default button rendering callback for single button.
      *
-     * @param string $name              Button name as it's written in template
-     * @param string $iconName          The part of Bootstrap glyphicon class that makes it unique
-     * @param array  $additionalOptions Array of additional options
-     *
+     * @param string $name Button name as it's written in template
+     * @param string $iconName The part of Bootstrap glyphicon class that makes it unique
+     * @param array $additionalOptions Array of additional options
      * @since 2.0.11
      */
-    protected function initDefaultButton($name, $iconName, $additionalOptions = [])
+    protected function initDefaultButton($name, $iconName, $additionalOptions = []): void
     {
-        if (!isset($this->buttons[$name]) && strpos($this->template, '{'.$name.'}') !== false) {
+        if (!isset($this->buttons[$name]) && strpos($this->template, '{' . $name . '}') !== false) {
             $this->buttons[$name] = function ($url, $model, $key) use ($name, $iconName, $additionalOptions) {
+                // TODO fix
```

Comment:
?

Reply:
We needs to pass a `language` from application (or from somewhere else) to these widgets. But I don't know how to do it at now.